devnet-sdk/systemext/l2.go

```diff
@@ -0,0 +1,148 @@
+
+			elRPC, err := findProtocolService(setup, ELServiceName, RPCProtocol, node.Services)
+			require.NoError(setup.T, err)
+			elClient := rpcClient(setup, elRPC)
+			l2.AddL2ELNode(system2.NewL2ELNode(system2.L2ELNodeConfig{
+				ELNodeConfig: system2.ELNodeConfig{
+					CommonConfig: commonConfig,
+					Client:       elClient,
+					ChainID:      l2ID,
+				},
+				ID: ids.EL,
+			}))
+
+			clRPC, err := findProtocolService(setup, CLServiceName, RPCProtocol, node.Services)
+			require.NoError(setup.T, err)
+			clClient := rpcClient(setup, clRPC)
+			l2.AddL2CLNode(system2.NewL2CLNode(system2.L2CLNodeConfig{
+				ID:           ids.CL,
+				CommonConfig: commonConfig,
+				Client:       clClient,
+			}))
+		}
+
+		for name, wallet := range net.Wallets {
+			priv, err := crypto.HexToECDSA(wallet.PrivateKey)
+			require.NoError(setup.T, err)
+			l2.AddUser(system2.NewUser(system2.UserConfig{
+				CommonConfig: commonConfig,
+				ID:           system2.UserID{Key: name, ChainID: l2ID},
+				Priv:         priv,
+			}))
+		}
+
+		setup.System.AddL2Network(l2)
+	}
+}
+
+func WithBatcher(idx int, l2ID system2.L2NetworkID, id system2.L2BatcherID) system2.Option {
+	return func(setup *system2.Setup) {
+		commonConfig := setup.CommonConfig()
+		env := getOrchestrator(setup).env
+		net := env.L2[idx]
+
+		l2 := setup.System.L2Network(l2ID)
+
+		batcherRPC, err := findProtocolService(setup, "batcher", RPCProtocol, net.Services)
+		require.NoError(setup.T, err)
+		l2.(system2.ExtensibleL2Network).AddL2Batcher(system2.NewL2Batcher(system2.L2BatcherConfig{
+			CommonConfig: commonConfig,
+			ID:           id,
+			Client:       rpcClient(setup, batcherRPC),
+		}))
+	}
+}
+
+func WithProposer(idx int, l2ID system2.L2NetworkID, id system2.L2ProposerID) system2.Option {
+	return func(setup *system2.Setup) {
+		commonConfig := setup.CommonConfig()
+		env := getOrchestrator(setup).env
+		net := env.L2[idx]
+
+		l2 := setup.System.L2Network(l2ID)
+
+		proposerRPC, err := findProtocolService(setup, "proposer", RPCProtocol, net.Services)
+		require.NoError(setup.T, err)
+		l2.(system2.ExtensibleL2Network).AddL2Proposer(system2.NewL2Proposer(system2.L2ProposerConfig{
+			CommonConfig: commonConfig,
+			ID:           id,
+			Client:       rpcClient(setup, proposerRPC),
+		}))
+	}
+}
+
+func WithChallenger(idx int, l2ID system2.L2NetworkID, id system2.L2ChallengerID) system2.Option {
+	return func(setup *system2.Setup) {
+		commonConfig := setup.CommonConfig()
+		env := getOrchestrator(setup).env
+		net := env.L2[idx]
+
+		l2 := setup.System.L2Network(l2ID)
+
+		_, err := findProtocolService(setup, "challenger", MetricsProtocol, net.Services)
+		require.NoError(setup.T, err)
+		l2.(system2.ExtensibleL2Network).AddL2Challenger(system2.NewL2Challenger(system2.L2ChallengerConfig{
+			CommonConfig: commonConfig,
+			ID:           id,
+		}))
+	}
+}
+
+func defineSystemKeys(setup *system2.Setup) system2.L2Keys {
+	// TODO: get actual mnemonic from Kurtosis
```

Comment:
```suggestion
	// TODO(#15040): improve key-loading from Kurtosis
```

Using the mnemonic is fine for now, but not what I'd like to do long-term. I opened an issue to track this work: https://github.com/ethereum-optimism/optimism/issues/15040